### ISSUE TITLE
Support for custom payload

### ISSUE THREAD

**Issue by hkroger** [outworkers/phantom#828]
Hi,

I didn't find any mention of custom payload in the code. Maybe it was an oversight but are there plans to add it? We would like to add custom payloads for end-to-end tracing.

**Comment by alexflav23** [comment]
Hi @hkroger What do you mean by custom payloads?

**Comment by alexflav23** [comment]
HI @hkroger We are going to add this in the next release, I'll keep you up to date on the timeline.

**Comment by hkroger** [comment]
Cool! Looking forward to this!

**Comment by alexflav23** [comment]
Hi @hkroger a bit overdue but:

```scala
 it should "allow setting metadata using a payload" in {
    val row = gen[TimeUUIDRecord].copy(id = UUIDs.timeBased())
    val payload = Payload("test" -> 5)
     val chain = for {
      queryRes <- database.timeuuidTable.store(row)
        .withOptions(_.outgoingPayload_=(payload))
        .future()
    } yield queryRes
     whenReady(chain) { res =>
      res.wasApplied() shouldEqual true
    }
  }
```

This is the PR, watch out for it when it goes in today/tomorrow, and the feature will become available in Phantom 2.24.11(I think that's the version number I might be wrong )

https://github.com/outworkers/phantom/pull/842/files
